### PR TITLE
[IOTDB-1634][IOTDB-1571] Extend more aligned timeseries insert APIs && Fix aligned timeseries wal cannot recover

### PR DESCRIPTION
**Pull request by HTHou**
## Description
Implemented the following session APIs. Added the usage in `AlignedTimeseriesSessionExample`.

- [x] Session
   - [x] insertAlignedRecords
   - [x] insertAlignedStringRecord
   - [x] insertAlignedStringRecords
   - [x] insertAlignedRecordsOfOneDevice
   - [x] insertTablets support insert tablet with aligned timeseries
- [x] SessionPool
   - [x] insertAlignedRecord
   - [x] insertAlignedRecords
   - [x] insertAlignedStringRecord
   - [x] insertAlignedStringRecords
   - [x] insertAlignedRecordsOfOneDevice
   - [x] insertTablet support insert tablet with aligned timeseries
   - [x] insertTablets support insert tablet with aligned timeseries

This PR is also fixing IOTDB-1571, #3760.

<hr>

### CODE REVIEW

**Review by HeimingZ — COMMENTED** [review]
1. There are some errors in javadoc, insertRecords should be replaced by insertAlignedRecords.
2. tablets in insertTablets method may empty, it is unsuitable to use get(0).

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 1]
```diff
@@ -861,25 +861,31 @@ public void insertRecord(
...
     insertRecord(deviceId, request);
   }
 
-  public void insertRecord(
-      String deviceId,
+  /**
+   * insert aligned data in one row, if you want improve your performance, please use
+   * insertAlignedRecords method or insertTablet method.
+   *
+   * @see Session#insertRecords(List, List, List, List, List)
```
Comment (HeimingZ) [inline session/src/main/java/org/apache/iotdb/session/Session.java:876]: Does insertRecords mean insertAlignedRecords?

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 2]
```diff
@@ -912,17 +918,37 @@ public void insertRecord(
...
-        genTSInsertStringRecordReq(deviceId, time, measurements, values);
+        genTSInsertStringRecordReq(deviceId, time, measurements, values, false);
     insertRecord(deviceId, request);
   }
 
+  /**
+   * insert aligned data in one row, if you want improve your performance, please use
+   * insertAlignedRecords method or insertTablet method.
+   *
+   * @see Session#insertRecords(List, List, List, List, List)
```
Comment (HeimingZ) [inline session/src/main/java/org/apache/iotdb/session/Session.java:929]: Does insertRecords mean insertAlignedRecords?

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java [hunk 3]
```diff
@@ -648,6 +777,37 @@ public void insertRecord(
     }
   }
 
+  /**
+   * insert aligned data in one row, if you want improve your performance, please use insertRecords
+   * method or insertTablet method.
+   *
+   * @see Session#insertRecords(List, List, List, List, List)
```
Comment (HeimingZ) [inline session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java:784]: Does insertRecords mean insertAlignedRecords?

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 4]
```diff
@@ -1425,7 +1587,11 @@ private TSInsertTabletsReq genTSInsertTabletsReq(List<Tablet> tablets, boolean s
       throws BatchExecutionException {
     TSInsertTabletsReq request = new TSInsertTabletsReq();
 
+    boolean isFirstTabletAligned = tablets.get(0).isAligned();
```
Comment (HeimingZ) [inline session/src/main/java/org/apache/iotdb/session/Session.java:1590]: IndexOutOfBoundsException occurs when tablets are empty

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java [hunk 5]
```diff
@@ -675,6 +835,33 @@ public void insertRecord(
     }
   }
 
+  /**
+   * insert aligned data in one row, if you want improve your performance, please use
+   * insertAlignedRecords method or insertTablet method.
+   *
+   * @see Session#insertRecords(List, List, List, List, List)
```
Comment (HeimingZ) [inline session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java:842]: Does insertRecords mean insertAlignedRecords?

File: session/src/main/java/org/apache/iotdb/session/Session.java (hunk 1 above)
Comment (HTHou) [reply session/src/main/java/org/apache/iotdb/session/Session.java]: Fixed

File: session/src/main/java/org/apache/iotdb/session/Session.java (hunk 2 above)
Comment (HTHou) [reply session/src/main/java/org/apache/iotdb/session/Session.java]: Fixed

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java (hunk 3 above)
Comment (HTHou) [reply session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java]: Fixed

**Comment by HeimingZ** [comment]
LGTM.

File: session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java (hunk 5 above)
Comment (HTHou) [reply session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java]: fixed

**Review by HeimingZ — APPROVED** [review]

File: session/src/main/java/org/apache/iotdb/session/Session.java (hunk 4 above)
Comment (HTHou) [reply session/src/main/java/org/apache/iotdb/session/Session.java:1597]: Fixed

**Pull request merged**